### PR TITLE
OCPQE-13004:Hypershift Agent workflow

### PR DESCRIPTION
**Pull request by LiangquanLi930**
Create hypershift cluster on Bare-Metal (Assistant install)

### CODE REVIEW

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-baremetalds-assisted-ovn-hypershift-agent-mgmt-p2-f7

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-baremetalds-assisted-ovn-hypershift-agent-mgmt-p2-f7

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-baremetalds-assisted-ovn-hypershift-agent-mgmt-p2-f7

**Comment by LiangquanLi930** [comment]
@jianlinliu Could you help to review this PR? Thanks

File: ci-operator/step-registry/cucushift/installer/rehearse/baremetalds/assisted/ovn/hypershift-agent/provision/cucushift-installer-rehearse-baremetalds-assisted-ovn-hypershift-agent-provision-chain.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+chain:
+  as: cucushift-installer-rehearse-baremetalds-assisted-ovn-hypershift-agent-provision
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/baremetalds/assisted/ovn/hypershift-agent/provision/cucushift-installer-rehearse-baremetalds-assisted-ovn-hypershift-agent-provision-chain.yaml:2]: I did not see a dedicated setting for `ovn`, so I guess we should rename the folder path to `cucushift/installer/rehearse/baremetalds/assisted/hypershift-agent/provision` ?

Comment (LiangquanLi930) [reply ci-operator/step-registry/cucushift/installer/rehearse/baremetalds/assisted/ovn/hypershift-agent/provision/cucushift-installer-rehearse-baremetalds-assisted-ovn-hypershift-agent-provision-chain.yaml:2]:
```
workflow:
  as: cucushift-installer-rehearse-baremetalds-assisted-ovn
  steps:
    allow_best_effort_post_steps: true
    allow_skip_on_success: true
    pre:
    - chain: cucushift-installer-rehearse-baremetalds-assisted-provision
    test:
    - ref: cucushift-installer-check
    post:
    - chain: cucushift-installer-rehearse-baremetalds-assisted-deprovision
    env:
      DEVSCRIPTS_CONFIG: |
        IP_STACK=v4
        NETWORK_TYPE=OVNKubernetes
        EXTRA_WORKER_VCPU=8
        EXTRA_WORKER_MEMORY=16384
        EXTRA_WORKER_DISK=100
        PROVISIONING_NETWORK_PROFILE=Disabled
        REDFISH_EMULATOR_IGNORE_BOOT_DEVICE=True
```
In our workflow, the mgmt cluster is OVN, so I have added the name of this chain.

File: ci-operator/step-registry/cucushift/installer/rehearse/baremetalds/assisted/ovn/hypershift-agent/deprovision/cucushift-installer-rehearse-baremetalds-assisted-ovn-hypershift-agent-deprovision-chain.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+chain:
+  as: cucushift-installer-rehearse-baremetalds-assisted-ovn-hypershift-agent-deprovision
+  steps:
+  - chain: hypershift-agent-destroy
+  - chain: cucushift-installer-rehearse-baremetalds-assisted-deprovision
+  documentation: |-
+    Destroy IPI cluster on Baremetal (via devscripts)
```
Comment (jhou1) [inline ci-operator/step-registry/cucushift/installer/rehearse/baremetalds/assisted/ovn/hypershift-agent/deprovision/cucushift-installer-rehearse-baremetalds-assisted-ovn-hypershift-agent-deprovision-chain.yaml:7]: Agent is not IPI, you could document it as baremetal install with assisted service

File: ci-operator/step-registry/cucushift/installer/rehearse/baremetalds/assisted/provision/cucushift-installer-rehearse-baremetalds-assisted-provision-chain.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+chain:
+  as: cucushift-installer-rehearse-baremetalds-assisted-provision
+  steps:
+  - chain: assisted-baremetal-operator
+  - ref: enable-qe-catalogsource
+  documentation: |-
+    Create an IPI cluster on Baremetal (via devscripts) for QE e2e tests.
```
Comment (jhou1) [inline ci-operator/step-registry/cucushift/installer/rehearse/baremetalds/assisted/provision/cucushift-installer-rehearse-baremetalds-assisted-provision-chain.yaml:7]: Same, assisted install is not IPI

File: ci-operator/step-registry/cucushift/installer/rehearse/baremetalds/assisted/ovn/hypershift-agent/OWNERS [hunk 4]
```diff
@@ -0,0 +1,14 @@
+approvers:
```
Comment (jhou1) [inline ci-operator/step-registry/cucushift/installer/rehearse/baremetalds/assisted/ovn/hypershift-agent/OWNERS:1]: Approvers/reviewers better be hypershift team

File: ci-operator/step-registry/cucushift/installer/rehearse/baremetalds/assisted/ovn/hypershift-agent/provision/cucushift-installer-rehearse-baremetalds-assisted-ovn-hypershift-agent-provision-chain.yaml (hunk 1 above)
Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/installer/rehearse/baremetalds/assisted/ovn/hypershift-agent/provision/cucushift-installer-rehearse-baremetalds-assisted-ovn-hypershift-agent-provision-chain.yaml:2]: I noticed the workflow explicitly set the OVN network type, but not in provision/deprovision chain, if you want to place the chains in ovn folder, I suggest to move the `env` to provision chain.

Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/installer/rehearse/baremetalds/assisted/ovn/hypershift-agent/provision/cucushift-installer-rehearse-baremetalds-assisted-ovn-hypershift-agent-provision-chain.yaml:2]: or only move provsion/deprovision chain to `cucushift/installer/rehearse/baremetalds/assisted/hypershift-agent/` folder, leave the workflow yaml file in the ovn folder.

Comment (LiangquanLi930) [reply ci-operator/step-registry/cucushift/installer/rehearse/baremetalds/assisted/ovn/hypershift-agent/provision/cucushift-installer-rehearse-baremetalds-assisted-ovn-hypershift-agent-provision-chain.yaml]: ok, I moved the workflow into hypershift dir to make it special for hypershift now.

**Comment by LiangquanLi930** [comment]
@jianlinliu @jhou1 Thanks for your review, Could you help to review this PR again?

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml [hunk 5]
```diff
@@ -2181,6 +2209,19 @@ tests:
     test:
     - chain: openshift-e2e-test-qe-destructive
     workflow: cucushift-installer-rehearse-azure-upi
+- as: baremetalds-assisted-ovn-hypershift-agent-mgmt-p2-f7
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml:2212]: Suggest to remove `ovn` accordingly in the job name.

Comment (LiangquanLi930) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml]: Thanks for your remind, Done

**Comment by LiangquanLi930** [comment]
@liangxia Could you help to review this PR? Thanks

**Comment by LiangquanLi930** [comment]
https://issues.redhat.com/browse/OCPQE-12317

**Pull request merged**